### PR TITLE
Add support for event language selection from config file

### LINKED ISSUE #14209: [Winlogbeat] Language setting for event log extraction
As per [Elastic form thread](https://discuss.elastic.co/t/forwarding-windows-event-logs-in-another-language/204067), Winlogbeat extracts Windows event logs in current system language, and this setting currently cannot be modified (some other threads suggested running the winlogbeat process as a user with the required language settings, but this does not change anything). It would be nice if the extracted log language was configurable.

The specific use case is server infrastructure with mixed system languages (in this particular case, some servers use English, others use a different language). All logs must be normalized into a single language, preferably English, and changing the system language for some servers is not an option. The problem cannot be solved by winlogbeat in its current form, and requires workarounds.

### PR DESCRIPTION
**Pull request by standa4**
## What does this PR do?

Add support for event language selection from config file - reaction to issue #14209
This change need to be verified by someone more experienced in go. I do **not** develop in go but the fix of the issue is important to me so I've tried to do as much as I could.
The setting of the language need to be specified for each type of log. For example edit 
```
winlogbeat.event_logs:
  - name: Application
    ignore_older: 72h

  - name: System
```
to
```
winlogbeat.event_logs:
  - name: Application
    ignore_older: 72h
    language: 0x0409

  - name: System
    language: 0x0409
```

`language: 0x0409` means English (United states) - (https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/a9eac961-e77d-41a6-90a5-ce1a8b0cdb9c)

## Why is it important?

This change allows to change the language of extracted events.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #14209

### CODE REVIEW

**Comment by kesslerm** [comment]
👍 reopening, as more users are asking for this enhancement through support tickets.

**Comment by standa4** [comment]
I've tried to merge my changes with latest commit in master branch but it need to be tested. I don't have my dev environment usable so I can't do it now. Maybe someone else could check it? Thanks

**Review by P1llus — APPROVED** [review]

**Comment by marc-gr** [comment]
@standa4 thanks for your contribution!

File: winlogbeat/docs/modules.asciidoc [hunk 1]
```diff
@@ -22,6 +22,9 @@ a `script` processor to your configuration file and point it at the included
 script file for the module. The documentation for each module includes an
 example.
 
+NOTE: The provided modules only support events in english. For more information
```
Comment (andrewkroh) [inline winlogbeat/docs/modules.asciidoc:25]:
```suggestion
NOTE: The provided modules only support events in English. For more information
```

Comment (andrewkroh) [reply winlogbeat/docs/modules.asciidoc:25]: Do we know what will break if a non-English language is used?

File: winlogbeat/docs/winlogbeat-options.asciidoc [hunk 2]
```diff
@@ -232,6 +232,20 @@ winlogbeat.event_logs:
 
 =======================================
 
+[float]
+==== `event_logs.language`
+
+The language ID the events will be rendered in. The language will be forced regardless
+of the system language. A complete list of language IDs can be found https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/a9eac961-e77d-41a6-90a5-ce1a8b0cdb9c[here].
+It defaults to `0`, which indicates to use the system language.
```
Comment (andrewkroh) [inline winlogbeat/docs/winlogbeat-options.asciidoc:240]:
```suggestion
of the system language. A complete list of language IDs can be found
https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/a9eac961-e77d-41a6-90a5-ce1a8b0cdb9c[here].
It defaults to `0`, which indicates to use the system language.
```

File: winlogbeat/docs/modules.asciidoc [hunk 3]
```diff
@@ -34,6 +37,7 @@ script processors that are guarded by a conditional `when` statement.
 winlogbeat.event_logs:
 - name: ForwardedEvents
   tags: [forwarded]
+  language: 0x0409
```
Comment (andrewkroh) [inline winlogbeat/docs/modules.asciidoc:40]:
```suggestion
  language: 0x0409 # en-US
```

I feel like a comment here would be helpful.

File: winlogbeat/docs/modules.asciidoc (hunk 1 above)
Comment (marc-gr) [reply winlogbeat/docs/modules.asciidoc:25]: I am doing some more testing around this, but from https://github.com/elastic/beats/issues/14209 and some support tickets, it seems separators and delimiters are different depending on the language, breaking the dissects in some cases.

File: winlogbeat/eventlog/wineventlog.go [hunk 4]
```diff
@@ -96,6 +97,7 @@ func (a NoMoreEventsAction) String() string { return noMoreEventsActionNames[a]
 // defaultWinEventLogConfig is the default configuration for new wineventlog readers.
 var defaultWinEventLogConfig = winEventLogConfig{
 	BatchReadSize: 100,
+	EventLanguage: 0,
```
Comment (efd6) [inline winlogbeat/eventlog/wineventlog.go:100]: Elide zero value initialisation?

**Comment by standa4** [comment]
Hello,
I noticed that assignment of default value for EventLanguage variable was lost during merge with upstream.
It should be set in file winlogbeat/eventlog/wineventlog.go within defaultWinEventLogConfig variable.
What is the proper way to fix this? New merge request or can you look at it yourselves?
Thank you.

**Comment by marc-gr** [comment]
> Hello,
> I noticed that assignment of default value for EventLanguage variable was lost during merge with upstream.
> It should be set in file winlogbeat/eventlog/wineventlog.go within defaultWinEventLogConfig variable.
> What is the proper way to fix this? New merge request or can you look at it yourselves?
> Thank you.

Thanks for taking a look! The `EventLanguage` property will default to `0`, so there is no need to explicitly set it, as pointed by the comment  in https://github.com/elastic/beats/pull/19818#discussion_r707906181. Is there anything in particular that should make us set it explicitly?

**Comment by standa4** [comment]
> > Hello,
> > I noticed that assignment of default value for EventLanguage variable was lost during merge with upstream.
> > It should be set in file winlogbeat/eventlog/wineventlog.go within defaultWinEventLogConfig variable.
> > What is the proper way to fix this? New merge request or can you look at it yourselves?
> > Thank you.
> 
> Thanks for taking a look! The `EventLanguage` property will default to `0`, so there is no need to explicitly set it, as pointed by the comment in [#19818 (comment)](https://github.com/elastic/beats/pull/19818#discussion_r707906181). Is there anything in particular that should make us set it explicitly?

Hello,
thank you for fast reply. You are right that it does not have to be set explicitly. As I wrote in the first post, I am not go developer. I have experience from other languages where variables need to be set with their initial values.
Sorry about my mistake. I need to get deeper knowledge of golang.

**Comment by marc-gr** [comment]
> Hello,
> thank you for fast reply. You are right that it does not have to be set explicitly. As I wrote in the first post, I am not go developer. I have experience from other languages where variables need to be set with their initial values.
> Sorry about my mistake. I need to get deeper knowledge of golang.

No need to apologize! Great work with the contribution 😄  Thanks!

**Pull request merged**